cartographer/mapping/map_builder_test.cc

```diff
@@ -111,21 +132,27 @@ class MapBuilderTest : public ::testing::Test {
   std::vector<::cartographer::transform::Rigid3d> local_slam_result_poses_;
 };
 
-TEST_F(MapBuilderTest, TrajectoryAddFinish2D) {
-  BuildMapBuilder();
-  int trajectory_id = map_builder_->AddTrajectoryBuilder(
-      {kRangeSensorId}, trajectory_builder_options_,
-      nullptr /* GetLocalSlamResultCallbackForSubscriptions */);
-  EXPECT_EQ(1, map_builder_->num_trajectory_builders());
-  EXPECT_TRUE(map_builder_->GetTrajectoryBuilder(trajectory_id) != nullptr);
-  EXPECT_TRUE(map_builder_->pose_graph() != nullptr);
-  map_builder_->FinishTrajectory(trajectory_id);
-  map_builder_->pose_graph()->RunFinalOptimization();
-  EXPECT_TRUE(map_builder_->pose_graph()->IsTrajectoryFinished(trajectory_id));
-}
+class MapBuilderTest : public MapBuilderTestBase<::testing::Test> {};
+class MapBuilderTestParamterizedByGridType
+    : public MapBuilderTestBase<::testing::TestWithParam<GridType>> {};
+class MapBuilderTestParamterizedByGridTypeAndDimensions
+    : public MapBuilderTestBase<
+          ::testing::TestWithParam<std::pair<GridType, int /* dimensions */>>> {
+};
+INSTANTIATE_TEST_CASE_P(MapBuilderTestParamterizedByGridType,
+                        MapBuilderTestParamterizedByGridType,
+                        ::testing::Values(GridType::PROBABILITY_GRID,
+                                          GridType::TSDF));
+INSTANTIATE_TEST_CASE_P(
+    MapBuilderTestParamterizedByGridTypeAndDimensions,
+    MapBuilderTestParamterizedByGridTypeAndDimensions,
+    ::testing::Values(std::make_pair(GridType::PROBABILITY_GRID, 2),
+                      std::make_pair(GridType::PROBABILITY_GRID, 3),
+                      std::make_pair(GridType::TSDF, 2)));
 
-TEST_F(MapBuilderTest, TrajectoryAddFinish3D) {
-  SetOptionsTo3D();
+TEST_P(MapBuilderTestParamterizedByGridTypeAndDimensions, TrajectoryAddFinish) {
+  if (GetParam().second == 3) SetOptionsTo3D();
```

Comment:
check if you can move these to the constructors of MapBuilderTestParamterizedByGridType and so on,
also for client_server_test if this works

Reply:
MapBuilderTestParamterizedByGridType ist a instantiation of the MapBuilderTestBase template, which is used for parameterized and non-parameterized tests. Therefore, the functions cannot be moved to the constructor.